src/sentry/static/sentry/app/views/eventsV2/results.tsx

```diff
@@ -293,6 +296,13 @@ class Results extends React.Component<Props, State> {
     }
   };
 
+  handleQueryLength = (value: string) => {
```

Comment:
I might be mistaken, but I think you can just derive `charsLeft` instead the smart search bar since you have the query length and max query length, so you don't need to use charsLeft anywhere.

Reply:
that cleaned things up a ton, thanks!